src/targets/services/receiveShares.js

```diff
@@ -0,0 +1,94 @@
+global.fetch = require('node-fetch').default
+global.btoa = require('btoa')
+
+import CozyClient from 'cozy-client'
+import log from 'cozy-logger'
+
+export const receiveShares = async () => {
+  // Worker's arguments
+  const {
+    docId,
+    sharecode,
+    uri,
+    nbShares,
+    parent,
+    finalize,
+    level,
+    aggregatorId,
+    executionId,
+    nbChild
+  } = JSON.parse(process.env['COZY_PAYLOAD'] || {})
+
+  const client = CozyClient.fromEnv(process.env, {})
+
+  const infoTag = 'info: [' + client.stackClient.uri.split('/')[2] + ']'
+
+  log(infoTag, 'Received share')
+
+  // Download share using provided informations
+  const sharedClient = new CozyClient({
+    uri: uri,
+    token: sharecode,
+    schema: {
+      files: {
+        doctype: 'io.cozy.files',
+        relationships: {
+          old_versions: {
+            type: 'has-many',
+            doctype: 'io.cozy.files.versions'
+          }
+        }
+      }
+    },
+    store: false
+  })
+  const share = await sharedClient.stackClient.fetchJSON(
+    'GET',
+    `/files/download/${docId}`
+  )
```

Comment:
This could be replaced by `const share = await client.collection('io.cozy.files').download(s)` : https://github.com/cozy/cozy-client/blob/master/packages/cozy-stack-client/src/FileCollection.js#L518

Reply:
Replacing an providing a io.cozy.files document as parameters give an error. The code seems to create  a download button and click it, which is a behavior only desirable in the browser, not a service

Reply:
Ah I see, this is because of https://github.com/cozy/cozy-client/blob/master/packages/cozy-stack-client/src/utils.js#L61

Then I guess [fetchFileContentById](https://github.com/cozy/cozy-client/blob/master/packages/cozy-stack-client/src/FileCollection.js#L553) would do the job, but it requires the filename, that you don't have it here, right?